releasenotes-builder/src/main/java/com/github/checkstyle/TemplateProcessor.java

```diff
@@ -52,20 +54,34 @@ private TemplateProcessor() { }
      * @param variables the map which represents template variables.
      * @param outputFile output file.
      * @param templateFilename template name.
+     * @param externalTemplate {@code true} if the template is an external file.
      * @throws IOException if I/O error occurs.
      * @throws TemplateException if an error occurs while generating freemarker template.
      */
     public static void generateWithFreemarker(Map<String, Object> variables, String outputFile,
-            String templateFilename) throws IOException, TemplateException {
+            String templateFilename, boolean externalTemplate) throws IOException,
+            TemplateException {
 
         final Configuration configuration = new Configuration(Configuration.VERSION_2_3_22);
         configuration.setDefaultEncoding("UTF-8");
         configuration.setLocale(Locale.US);
         configuration.setTemplateExceptionHandler(TemplateExceptionHandler.RETHROW_HANDLER);
         configuration.setNumberFormat("0.######");
-        configuration.setClassForTemplateLoading(Main.class, "/" + TEMPLATE_FOLDER_PATH);
 
-        final Template template = configuration.getTemplate(templateFilename);
+        final String fileName;
+
+        if (externalTemplate) {
+            final File file = new File(templateFilename);
+
+            fileName = file.getName();
+            configuration.setTemplateLoader(new FileTemplateLoader(file.getParentFile()));
+        }
+        else {
+            fileName = templateFilename;
+            configuration.setClassForTemplateLoading(Main.class, "/" + TEMPLATE_FOLDER_PATH);
```

Comment:
as you move loading of tempalate of out this class, you will resolve cycle dependency of Main and TemplateProcessor.

Reply:
Done.